makefile

```diff
@@ -64,21 +64,31 @@ CWARNS= $(CWARNSCPP) $(CWARNSC) $(CWARNGCC)
 
 LOCAL = $(TESTS) $(CWARNS)
 
-
 # enable Linux goodies
-MYCFLAGS= $(LOCAL) -std=c99 -DLUA_USE_LINUX -DLUA_USE_READLINE
-MYLDFLAGS= $(LOCAL) -Wl,-E
-MYLIBS= -ldl -lreadline
-
-
-CC= gcc
-CFLAGS= -Wall -O2 $(MYCFLAGS) -fno-stack-protector -fno-common -march=native
-AR= ar rc
-RANLIB= ranlib
-RM= rm -f
-
-
-
+MYCFLAGS  = $(LOCAL) -DLUA_USE_LINUX -DLUA_USE_READLINE
+MYLIBS    = -ldl -lreadline
+MYLDFLAGS = $(LOCAL) -Wl,-E
+RM        = rm -f
+BD        = cmp/$(GWC_ARCH)
```

Comment:
Maybe BD could be just "./" and then `ifdef GWC_ARCH` you make it `cmp/$(GWC_ARCH)` ?

Reply:
Something like:
```
diff --git a/makefile b/makefile
index b119c5e9..bcd8d476 100644
--- a/makefile
+++ b/makefile
@@ -69,9 +69,10 @@ MYCFLAGS  = $(LOCAL) -DLUA_USE_LINUX -DLUA_USE_READLINE
 MYLIBS    = -ldl -lreadline
 MYLDFLAGS = $(LOCAL) -Wl,-E
 RM        = rm -f
-BD        = cmp/$(GWC_ARCH)
+BD        = ./
 
 ifdef GWC_ARCH
+BD        = cmp/$(GWC_ARCH)
 # Our custom setting when running under cshell
 ifeq ($(GWC_ARCH),ARM_V3)
 MYCFLAGS = $(LOCAL) -DLUA_USE_LINUX  -Wno-redundant-decls

```

Reply:
Yess, that is exactly what I was going to propose to you.
Now, non `GWC_ARCH` builds are going to save the output in `cmp/lua`

Of course I can leave it in `./` if you like, but probably you'll need to update the yocto recipe anyway, as it was previously using the tarball that has a different structure from the github source code. Don't you?

Reply:
From a fast look the only difference should be a "src/" mode, so just putting as yocto base directory += "src/" should solve all the rest. But yes, in principle we will need to touch the recipe at least a bit, but maybe the less the better :-) (also for future upgrades)

Reply:
would it be good to gust add a `+= cmp/` :wink:  so I can avoid having `.o` mixed with the source code.

Anyway. I'm going to delete this branch and reapply everything as we need to go back to v5.3.5. ok?